File: perl/p1.pl
#!/usr/bin/perl

sub SumDivisible {
  $p = int($_[1] / $_[0]);
  return (int($_[0] * ($p *( $p + 1 )) / 2));
}

$max = 999;

if (scalar(@ARGV) > 0){
  $max=$ARGV[0] - 1;
}

$sum = SumDivisible(3, $max) + SumDivisible(5, $max) - SumDivisible(15, $max);
printf $sum;

File: python3/p1.py
#!/usr/bin/python3

import sys

def SumDivisible(n, m):
    p = m // n;
    return n * (p * (p + 1)) // 2

def main():
    m = 999
    if len(sys.argv) > 1:
        m = int(sys.argv[1])
    s = SumDivisible(3, m) + SumDivisible(5, m) - SumDivisible(15, m)
    print(s)

if __name__ == "__main__":
    main()
